webpack/assets/javascripts/react_app/components/Pagination/Pagination.js

```diff
@@ -21,11 +25,12 @@ const Pagination = props => {
     dropdownButtonId,
     disableNext,
     disablePrev,
+    isPF4,
```

Comment:
doesn't this mean it will still default to pf3? Why not modify the implementation to only use pf4 pagination instead of having this flag to opt-in?

Reply:
by default the prop value is set to `true` in the [defaultProps](https://github.com/theforeman/foreman/pull/8990/files#diff-e807f39d86a92373df27f9266c31f89c99c99f57a9cdc0552db185497d3bbe23R130)
I believe it could be OK to have a small period where both paginations are available and then we could remove the PF3 implementation.

since `usePaginationOptions` is also affected in this PR and consumed by plugins, it will make more sense to do it separately imho

Reply:
Ah, missed the default prop. 
But I think the change there would actually break plugins using it - see for example here:

https://github.com/theforeman/foreman_openscap/blob/a53b008174e96578f4cd07e6ef61c94e60804020/webpack/components/IndexTable/index.js#L26-L39

The pagination options would be set using pf3 while the pagination component would be set to pf4. 
TBH, I'm not sure I understand why that hook should be used anywhere outside this component (or exist as a hook), it should be an internal implementation detail...